src/build/patches.ts

```diff
@@ -135,13 +160,58 @@ function handleEvent(child: Node): Event {
  */
 function handleProperty(child: Node): Partial<Property> {
   return {
-    name: child.values[0] as string,
+    name: string(child.values[0]),
     ...optionalMember("exposed", "string", child.properties?.exposed),
     ...optionalMember("optional", "boolean", child.properties?.optional),
     ...optionalMember("overrideType", "string", child.properties?.overrideType),
   };
 }
 
+/**
+ * Handles a child node of type "method" and adds it to the method object.
+ * @param child The child node to handle.
+ */
+function handleMethod(child: Node): Partial<Method> {
+  const name = string(child.values[0]);
+  const returnType = child.properties.returns;
+
+  let typeNode: Node | undefined;
+  const params: { name: string; type: string }[] = [];
+
+  for (const c of child.children) {
+    switch (c.name) {
+      case "type":
+        if (typeNode) {
+          throw new Error(`Method "${name}" has multiple type nodes (invalid)`);
```

Comment:
I actually want to allow this for union types, but for now this is ok as we don't have union types right now